xarray/core/utils.py

```diff
@@ -183,7 +183,7 @@ def is_full_slice(value):
     return isinstance(value, slice) and value == slice(None)
 
 
-def combine_pos_and_kw_args(pos_kwargs, kw_kwargs, func_name):
+def either_dict_or_kwargs(pos_kwargs, kw_kwargs, func_name):
```

Comment:
We don't need to test every use, but we should at least add unit test for this helper function -- I don't think we have any test coverage currently.